extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/LayoutPresenterCompilerStates.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.agera.rvadapter;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import com.google.android.agera.Receiver;
+
+/**
+ * Container of the compiler state interfaces supporting the creation of a {@link
+ * LayoutPresenter}.
+ */
+public interface LayoutPresenterCompilerStates {
```

Comment:
another idea: the RP / DBRP / LP compilers don't need to handle complex Java type safety, so they can all be simplified to builder model (see RA.Builder, a public final class instead of a collection of interfaces).